src/api/v4/dataview/base.js

```diff
@@ -0,0 +1,227 @@
+var _ = require('underscore');
+var Backbone = require('backbone');
+var STATUS = require('../constants').STATUS;
+var SourceBase = require('../source/base');
+
+/**
+ * Base dataview object
+ *
+ * @constructor
+ * @abstract
+ * @memberof carto.dataview
+ * @api
+ */
+function Base () {}
+
+_.extend(Base.prototype, Backbone.Events);
+
+/**
+ * Return the current dataview status
+ *
+ * @return {carto.dataview.STATUS} Current dataview status
+ * @api
+ */
+Base.prototype.getStatus = function () {
+  return this._status;
+};
+
+/**
+ * Return true is the current status is loading
+ *
+ * @return {boolean}
+ * @api
+ */
+Base.prototype.isLoading = function () {
+  return this._status === STATUS.LOADING;
+};
+
+/**
+ * Return true is the current status is loaded
+ *
+ * @return {boolean}
+ * @api
+ */
+Base.prototype.isLoaded = function () {
+  return this._status === STATUS.LOADED;
+};
+
+/**
+ * Return true is the current status is error
+ *
+ * @return {boolean}
+ * @api
+ */
+Base.prototype.hasError = function () {
+  return this._status === STATUS.ERROR;
+};
+
+/**
+ * Enable the dataview
+ *
+ * @return {carto.dataview.Base} this
+ * @api
+ */
+Base.prototype.enable = function () {
+  return this._setEnabled(true);
+};
+
+/**
+ * Disable the dataview
+ *
+ * @return {carto.dataview.Base} this
+ * @api
+ */
+Base.prototype.disable = function () {
+  return this._setEnabled(false);
+};
+
+/**
+ * Return true if the dataview is enabled
+ *
+ * @return {boolean}
+ * @api
+ */
+Base.prototype.isEnabled = function () {
+  return this._enabled;
+};
+
+/**
+ * Set the dataview column
+ *
+ * @param  {string} column
+ * @return {carto.dataview.Base} this
+ * @api
+ */
+Base.prototype.setColumn = function (column) {
+  this._checkColumn(column);
+  this._column = column;
+  if (this._internalModel) {
+    this._internalModel.set('column', this._column);
+  }
+  return this;
+};
+
+/**
+ * Return the current dataview column
+ *
+ * @return {string} Current dataview column
+ * @api
+ */
+Base.prototype.getColumn = function () {
+  return this._column;
+};
+
+Base.prototype.getData = function () {
+  throw new Error('getData must be implemented by the particular dataview.');
+};
+
+// syncOnBoundingChanges
+// isSyncedOnBoundingChanges
+
+// Protected methods
+
+Base.prototype._initialize = function (source, column, options) {
+  options = this._defaultOptions(options);
+
+  this._checkSource(source);
+  this._checkColumn(column);
+  this._checkOptions(options);
+
+  this._source = source;
```

Comment:
Sime inline comments explaining those fields would be great for newcomers in the dataviews world like me ^^

Reply:
Added doc in the functions' header